webui/react/src/components/JupyterLabModal.tsx

```diff
@@ -317,7 +317,9 @@ const JupyterLabFullConfig: React.FC<FullConfigProps> = ({
             },
           ]}>
           <CodeEditor
-            files={[{ content: config, key: 'config.yaml' }]}
+            files={[
+              { get: () => Promise.resolve(Loadable.getOrElse('', config)), key: 'config.yaml' },
```

Comment:
Shouldn't you be using observables `toPromise` here so that the promise resolves when the file is loaded?

Reply:
Something like `config.onlyIf(Loadable.isLoaded).toPromise()`?

Reply:
i'll tinker around with this -- i think there's edge cases around `toPromise` only resolving when the value actually changes that we'll need to handle.

Reply:
We might need to create a little `resolveWhen: (Observable<T>, T => boolean) => Promise<T>` utility in `observable.ts`. It should be pretty straightforward to define.

Reply:
The more i'm thinking about it, maybe returning an observable promise isn't entirely necessary for this case?

While we do want the promise to be unresolved if the value isn't loaded, the file array will rerender (and get will be called again) whenever the config changes. So, even if the promise resolves, it's just going to be overwritten with another resolved promise when the underlying value changes.

Reply:
If that's the case why not just accept a `Loadable<file>`?

Reply:
okay, moved file loading into the experiment code view instead.